### PR TITLE
Update web3/sessions to become web3/auth

### PR DESCRIPTION
**Pull request by gbibeaul**
### Description

The sessions example was not super clear on what it helped people achieve. This is a more to the point example on a pain point experiences by multiple web3 projects we got in touch with.

Note: There is a usecase for a session without login with ethereum. To keep an healthy selection of web3 content I'll keep both after talking with Steven.

### Best Way to Test

1. Navigate to the [example](https://web3-authentication.vercel.app)
2. If you don't have metamask you will see a message asking you to install it
3. if you have metamask you will be able to login and logout
4. see that the address name updates

### Type of Change

- [x] example improvement

### New Example Checklist

- [ ] 🚀 Link to deployment URL on Vercel
- [ ] 📱 Is it responsive? Are mobile and tablets considered?
- [ ] 📚 `README.md` preview link in PR description (branch)
- [ ] ⚙️ Secrets have instructions for how to set up in the readme
- [ ] 🛫 `npm run new-example` was run to create the example

### CODE REVIEW

File: solutions/web3-authentication/README.md [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
 
-This example shows how create sessions in Next.js with [Metamask](https://metamask.io/) and [NextAuth.js](https://next-auth.js.org/).
+This example shows how to implement web3 authentication with Next.js
 
 ## Demo
 
-https://web3-sessions.vercel.app
-
-This Demo will show how to configure NextAuth.js and create a custom hook to protect routes.
+https://web3-authentication.vercel.app
```
Comment (goncy) [inline solutions/web3-authentication/README.md:7]: This should be prefixed with `solutions-`

File: solutions/web3-authentication/pages/api/hello.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import type { NextApiRequest, NextApiResponse } from 'next'
```
Comment (goncy) [inline solutions/web3-authentication/pages/api/hello.ts:1]: You can delete this file

File: solutions/web3-authentication/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+  const handleLogin = async () => {
+    try {
+      await connect(connectData.connectors[0])
+      const callbackUrl = '/protected'
+      const message = new SiweMessage({
+        domain: window.location.host,
+        address: accountData?.address,
+        statement: 'Sign in with Ethereum to the app.',
+        uri: window.location.origin,
```
Comment (goncy) [inline solutions/web3-authentication/pages/index.tsx:31]: Can we get this from the router? Just to not rely directly on window

**Review by goncy — CHANGES_REQUESTED** [review]

File: solutions/web3-authentication/pages/index.tsx (hunk 3 above)
Comment (gbibeaul) [reply solutions/web3-authentication/pages/index.tsx:31]: I checked the docs and it does not seem like it. 

We could have it come from a Env variable instead. What do you think?

**Review by goncy — APPROVED** [review]

**Pull request merged**